### PR TITLE
Introduce edit job modal

### PR DESCRIPTION
**Pull request by orlandohohmeier**
edit-modal
Introduce a job edit modal.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @pierlo-upitup

**Comment by orlandohohmeier** [comment]
Integration test are passing locally:
> `131 passing (4m)`
> `19 pending`

**Comment by rcorral** [comment]
Code looks good 💯

File: src/js/components/modals/JobFormModal.js [hunk 1]
```diff
@@ -224,11 +258,16 @@ class JobFormModal extends mixin(StoreMixin) {
 }
 
 JobFormModal.defaultProps = {
-  onClose: function () {},
+  isEdit: false,
+  job: new Job(),
+  onClose: function () {
```
Comment (aldipower) [inline src/js/components/modals/JobFormModal.js]: One-liner?

File: src/js/pages/jobs/JobDetailPage.js [hunk 2]
```diff
@@ -97,7 +103,7 @@ class JobDetailPage extends mixin(StoreMixin, TabsMixin) {
       <button
         className="button button-inverse button-stroke"
         key="edit"
-        onClick={this.handleEditButtonClick.bind(this, jobID)}>
+        onClick={this.handleEditButtonClick.bind(this)}>
```
Comment (aldipower) [inline src/js/pages/jobs/JobDetailPage.js]: Could you bind this in the constructor for consistency reasons?

File: src/js/pages/jobs/JobDetailPage.js [hunk 3]
```diff
@@ -221,7 +227,9 @@ class JobDetailPage extends mixin(StoreMixin, TabsMixin) {
           subTitle={this.getSubTitle(job)}
           subTitleClassName={{emphasize: false}}
           title={job.getDescription()} />
-          {this.tabs_getTabView(job)}
+        {this.tabs_getTabView(job)}
+        <JobFormModal isEdit={true} job={job} open={this.state.isJobFormModalOpen}
+          onClose={this.handleCloseJobFormModal.bind(this)} />
```
Comment (aldipower) [inline src/js/pages/jobs/JobDetailPage.js]: constructor bind

**Comment by aldipower** [comment]
Nice! Some minor comments.

File: src/js/pages/jobs/JobDetailPage.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/pages/jobs/JobDetailPage.js]: I actually thought about refactoring this, but as this PR is only about introducing the edit modal, I decided not to adjust these. I'd prefer to do this in another PR, are you ok with that?

File: src/js/components/modals/JobFormModal.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/components/modals/JobFormModal.js]: Done!

File: src/js/pages/jobs/JobDetailPage.js (hunk 2 above)
Comment (aldipower) [inline src/js/pages/jobs/JobDetailPage.js]: As this is a actually small change - isn't it? - I would prefer to have this in this PR.
But decide on your own. You could also create overhead and create a new PR.

File: src/js/pages/jobs/JobDetailPage.js [hunk 4]
```diff
@@ -57,26 +72,32 @@ class JobDetailPage extends mixin(StoreMixin, TabsMixin) {
...
+    let job = ChronosStore.getJob(this.props.params.id);
+
     if (selection.id === 'suspend') {
-      ChronosStore.suspendJob(jobID);
+      ChronosStore.suspendJob(job.getId());
     }
 
     if (selection.id === 'destroy') {
-      ChronosStore.deleteJob(jobID);
+      ChronosStore.deleteJob(job.getId());
```
Comment (pierlo-upitup) [inline src/js/pages/jobs/JobDetailPage.js]: I'd have preferred to see these various fixes in a separate PR but I won't complain 😂

**Comment by pierlo-upitup** [comment]
I'm merge-friendly!

File: src/js/pages/jobs/JobDetailPage.js (hunk 4 above)
Comment (orlandohohmeier) [inline src/js/pages/jobs/JobDetailPage.js]: https://github.com/dcos/dcos-ui/pull/464#discussion_r68016172
